src/main/java/com/spotify/folsom/client/Request.java

```diff
@@ -65,4 +72,32 @@ protected static ByteBuf toBuffer(final ByteBufAllocator alloc, ByteBuffer dst,
   public int getOpaque() {
     return opaque;
   }
+
+  @VisibleForTesting
+  static byte[] encodeKey(String key, Charset charset) {
+    checkNotNull(key, "key");
+    byte[] keyBytes = key.getBytes(charset);
+    int length = keyBytes.length;
+    if (length > 250) {
+      throw new IllegalArgumentException("Key is too long: " + key);
+    }
+    if (length <= 0) {
+      throw new IllegalArgumentException("Key is empty");
+    }
+    for (int i = 0; i < length; i++) {
+      final byte c = keyBytes[i];
+      if (c >= 0 && c <= 32) {
+        throw new IllegalArgumentException("Invalid key: " + key);
```

Comment:
It does indeed seem like memcached just looks for 0x20 (' ') to tokenize things, so a key may not contain that character. It also uses '\0' and '\r' and '\n' so those are probably also a bad idea to use. We could relax our current key byte requirement from [33, 255] to everything except 0, \r, \n and ' ', but I am not sure how much that actually improves things. We still won't be able to claim that we support encodings such as UTF-16

https://github.com/memcached/memcached/blob/master/memcached.c#L2443